src/to-have-text-content.js

```diff
@@ -1,11 +1,15 @@
-import {checkHtmlElement, getMessage, matches, normalize} from './utils'
+import {getMessage, HtmlElementTypeError, matches, normalize} from './utils'
 
 export function toHaveTextContent(
   htmlElement,
   checkWith,
   options = {normalizeWhitespace: true},
 ) {
-  checkHtmlElement(htmlElement, toHaveTextContent, this)
+  const window = htmlElement.ownerDocument.defaultView
```

Comment:
There should be a check prior to this line. Otherwise this line can throw when receiving something that does not even have a `ownerDocument` property. Or if the obtained `window` value is not a window object and does not have a `window.Node` thing in it.